huaweicloud/services/ecs/resource_huaweicloud_compute_auto_launch_group.go

```diff
@@ -0,0 +1,469 @@
+			},
+			"launch_template_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"overrides": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"flavor_id": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"spot_price": {
+							Type:     schema.TypeFloat,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+						"priority": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+						"weighted_capacity": {
+							Type:     schema.TypeFloat,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					m := v.(map[string]interface{})
+					return hashcode.String(m["availability_zone"].(string) + m["flavor_id"].(string))
+				},
+			},
+			"delete_instances": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"stable_capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"excess_fulfilled_capacity_behavior": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instances_behavior_with_expiration": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"valid_since": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"valid_until": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"allocation_strategy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"supply_option": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"spot_price": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"current_capacity": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"current_stable_capacity": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+			},
+			"task_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAutoLaunchGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	product := "cms"
+	client, err := cfg.NewServiceClient(product, "")
```

Comment:
`region` should not be empty because the endpoint maybe cms.{region}.xxx.com in other clouds.
```suggestion
	client, err := cfg.NewServiceClient(product, region)
```